src/client/src/Stream.c

```diff
@@ -1836,6 +1842,167 @@ STATUS setNalAdaptationFlags(PKinesisVideoStream pKinesisVideoStream, UINT32 nal
+    StackQueueIterator iterator;
+    UINT64 data;
+
+    CHK(pKinesisVideoStream != NULL, STATUS_NULL_ARG);
+    pKinesisVideoClient = pKinesisVideoStream->pKinesisVideoClient;
+
+    // Check if the stream has been stopped
+    CHK(!pKinesisVideoStream->streamStopped, STATUS_STREAM_HAS_BEEN_STOPPED);
+
+    // Lock the stream
+    pKinesisVideoClient->clientCallbacks.lockMutexFn(pKinesisVideoClient->clientCallbacks.customData, pKinesisVideoStream->base.lock);
+    streamLocked = TRUE;
+
+    if(pMetadata == NULL) {
+        hasMetadata = FALSE;
+    }
+
+    if(pKinesisVideoStream->metadataTracker.activeMkvEvent)
+    {
+        //only the first event may have a L1 'tags' element.
+        firstEvent = FALSE;
+    }
+
+    // Check if we are in the right state only if we are not in a rotation state
+    if (pKinesisVideoStream->streamState != STREAM_STATE_READY) {
+        CHK_STATUS(acceptStateMachineState(pKinesisVideoStream->base.pStateMachine,
+                                           STREAM_STATE_READY | STREAM_STATE_PUT_STREAM | STREAM_STATE_TAG_STREAM | STREAM_STATE_STREAMING |
+                                               STREAM_STATE_GET_ENDPOINT | STREAM_STATE_GET_TOKEN | STREAM_STATE_STOPPED));
+    }
+
+    // Validate if the customer is not attempting to add an internal metadata
+    if(hasMetadata) {
+        for(UINT8 iter = 0; iter < pMetadata->numberOfPairs; iter++) {
+            CHK(0 != STRNCMP(AWS_INTERNAL_METADATA_PREFIX, pMetadata->names[iter], (SIZEOF(AWS_INTERNAL_METADATA_PREFIX) - 1) / SIZEOF(CHAR)), STATUS_INVALID_METADATA_NAME);
+        }
+        if(pMetadata->imagePrefix != NULL) {
+            CHK(0 != STRNCMP(AWS_INTERNAL_METADATA_PREFIX, pMetadata->imagePrefix, (SIZEOF(AWS_INTERNAL_METADATA_PREFIX) - 1) / SIZEOF(CHAR)), STATUS_INVALID_METADATA_NAME);
+        }
+    }
+
+    // Ensure we don't have more than MAX size of the metadata queue
+    switch(event) {
+        case STREAM_EVENT_NOTIFICATION:
+            neededNodes++;
+            break;
+        case STREAM_EVENT_IMAGE_GENERATION:
+            neededNodes++;
+            if(hasMetadata && pMetadata->imagePrefix != NULL) {
+                neededNodes++;
+            }
+            break;
+        default:
+            break;
+    }
+    if(hasMetadata) {
+        neededNodes += pMetadata->numberOfPairs;
+    }
+    CHK_STATUS(stackQueueGetCount(pKinesisVideoStream->pMetadataQueue, &metadataQueueSize));
+    CHK((metadataQueueSize + neededNodes) <= MAX_FRAGMENT_METADATA_COUNT, STATUS_MAX_FRAGMENT_METADATA_COUNT);
+
+    // reset iterator for size calculations
+    neededNodes = 0;
+
+
+    // Check whether we are OK to package the metadata but do not package.
+    // create all metadata nodes
+    creatingNodes = TRUE;
+    switch(event) {
+        case STREAM_EVENT_NOTIFICATION:
+            if(firstEvent) {
+                CHK_STATUS(mkvgenGenerateTagsChain(NULL, KVSEVENT_NOTIFICATION_STRING, "", &packagedSize, MKV_TREE_TAGS));
```

Comment:
I very much disagree that adding a ternary operator in function parameters improved readability over an external if/else.

Reply:
Perhaps. However, the repetition of code in these types of constructs makes the code very brittle and not future proof. You could see the entire codebase has similar constructs and it's there for a reason. 
An alternative might be to prepare the argument (with or without ternary operator) and then call the APIs without branching (aside from perf gains that non-branching provides)

Reply:
I'm going to have to agree with @MushMal that in this case using ternary operator is more readable.  I think when you have these code blocks repeated over and over the first thing one tends to do is to find out how they are not identical.  I think that is more apparent with the ternary operator since the only difference in this case is the MKV_TREE_* enum.  I would go further and say we should have a ternary operator for the the `KVSEVENT*` as well reducing two if/else pairs and 8 lines of code with 2 lines of code.

Reply:
You have to deal with the switch so 4 lines without if/else then, I still think that's a lot better than the identical if/elses its easy to miss withs different there since out of 5/6 parameters, only 1 is different between if and else.

Reply:
With the most recent change from enum-> bit shifted defines it became necessary to add another flag to be checked & changed, thus a ternary operator in the function is not longer sufficient.